zebra-rpc/src/methods.rs

```diff
@@ -1308,3 +1362,41 @@ fn check_height_range(start: Height, end: Height, chain_height: Height) -> Resul
 
     Ok(())
 }
+
+/// Given a potentially negative index, find the corresponding `Height`.
+///
+/// This function is used to parse the integer index argument of `get_block_hash`.
+fn get_height_from_int(index: i32, tip_height: Height) -> Result<Height> {
+    if index >= 0 {
+        let height = index.try_into().expect("Positive i32 always fits in u32");
+        if height > tip_height.0 {
+            return Err(Error::invalid_params(
+                "Provided index is greater than the current tip",
+            ));
+        }
+        Ok(Height(height))
+    } else {
+        let height = (tip_height.0 as i32).checked_add(index + 1);
```

Comment:
I opened PR #5428 with some suggestions to check or document the overflows in this code.